### PR TITLE
C#: Fix a URL redirection from remote source false positive

### LINKED ISSUE #14952: False positive: C# URL redirection from remote source
**Description of the false positive**

When guarding a redirect with `HttpRequestBase.IsUrlLocalToHost()`, CodeQL still flags the redirect. Other methods of checking whether the URL is local (i.e. `UrlHelper.IsLocalUrl()`) prevent CodeQL from flagging the redirect, so `HttpRequestBase.IsUrlLocalToHost()` should too.

**Code samples or links to source code**

A minimal ASP.NET Web Forms page that reproduces the false positive:

```csharp
using System;
using System.Web;
using System.Web.UI;
using System.Web.WebPages;

namespace WebForms
{
	public partial class Redirect : Page
	{
		protected void Page_Load(object sender, EventArgs e)
		{
			var isReturnUrlLocal = new HttpRequestWrapper(Request).IsUrlLocalToHost(Request.QueryString["ReturnUrl"]);
			if (isReturnUrlLocal)
				Response.Redirect(Request.QueryString["ReturnUrl"], false);
		}
	}
}
```

**URL to the alert on GitHub code scanning (optional)**

The alert is in a private repository.

### PR DESCRIPTION
**Pull request by rpmrmartin**
Fixes #14952

### CODE REVIEW

**Comment by michaelnebel** [comment]
DCA Looks good.
We probably need a second review.

File: csharp/ql/test/query-tests/Security Features/CWE-601/UrlRedirect/UrlRedirect.cs
Comment (hvitved) [inline csharp/ql/test/query-tests/Security Features/CWE-601/UrlRedirect/UrlRedirect.cs:3]: According to https://learn.microsoft.com/en-us/dotnet/api/system.web.webpages.requestextensions.isurllocaltohost?view=aspnet-webpages-3.2, the method lives in `System.Web.WebPages`.

**Review by hvitved — CHANGES_REQUESTED** [review]
Thanks for the fix; one minor comment.

File: csharp/ql/test/query-tests/Security Features/CWE-601/UrlRedirect/UrlRedirect.cs
Comment (michaelnebel) [reply csharp/ql/test/query-tests/Security Features/CWE-601/UrlRedirect/UrlRedirect.cs:3]: Yes, you are right - I will fix that.

**Review by hvitved — APPROVED** [review]

**Pull request merged**
